polaris-migrator/src/utilities/sass.ts

```diff
@@ -249,6 +297,33 @@ export function toTransformablePx(value: string) {
  */
 export const StopWalkingFunctionNodes = false;
 
+/**
+ * All transformable duration units. These values are used to determine
+ * if a decl.value can be mapped to a Polaris custom property.
+ *
+ * Note: <time> is a dimension with 's' or 'ms' as the unit:
+ * https://w3c.github.io/csswg-drafts/css-values-3/#time-value
+ */
+export const transformableDurationUnits = ['s', 'ms'];
+
+export function isTransformableDuration(
+  dimension: false | Dimension,
+): dimension is Dimension {
+  if (!dimension) return false;
+
+  // Zero is the only unitless dimension our duration transforms support
+  if (isUnitlessZero(dimension)) return true;
+
+  return transformableDurationUnits.includes(dimension.unit);
+}
+
+export function isPolarisVar(node: Node): boolean {
+  return (
+    isSassFunction('var', node) &&
+    (node.nodes?.[0]?.value ?? '').startsWith('--p-')
```

Comment:
Maybe we can be extra sure this is a Polaris token by leveraging this [RegEx logic here](https://github.com/Shopify/polaris/pull/7375/files#diff-3ed201423f7ef97686e85c262e4c5adf310c70a0576e6c9c51a87a6887465620R242-R244)

Reply:
Ah, good point: I forgot about 3P devs! I've made that change 👍